assets/src/demo/app.js

```diff
@@ -0,0 +1,188 @@
+		<h2>Datetimes List</h2>
+		<ul>{ getDatetimesList() }</ul>
+	</div>;
+};
+
+const DatetimesList = withDispatch( ( dispatch, ownProps, { select } ) => {
+	async function onDateClone( datetime ) {
+		const { getRelatedEntities } = select( 'eventespresso/core' );
+		const { createEntity, createRelations } = dispatch( 'eventespresso/core' );
+		const tickets = getRelatedEntities( datetime, 'tickets' );
+		const newDateTime = await createEntity( 'datetime', datetime.forClone );
+		createRelations(
+			'datetime',
+			newDateTime.id,
+			'tickets',
+			tickets
+		);
+		createRelations(
+			'event',
+			ownProps.eventId,
+			'datetimes',
+			[ newDateTime ]
+		);
+	}
+	return { onDateClone };
+} )( DatesList );
+
+const TicketsList = ( { tickets } ) => {
+	function getTicketsList() {
+		if ( tickets.length > 0 ) {
+			return tickets.map( ( ticket ) => <li key={ ticket.id }>{ ticket.name }</li> );
+		}
+		return <li><Spinner /></li>;
+	}
+	return <div>
+		<h2>Tickets List</h2>
+		<ul>{ getTicketsList() }</ul>
+	</div>;
+};
+
+const DateAndTicketList = ( {
+	eventId,
+	datetimes = [],
+	tickets = [],
+	isInitializing,
+} ) => {
+	function getContent() {
+		return isInitializing ?
+			<Spinner /> :
+			<Fragment>
+				<DatetimesList eventId={ eventId } datetimes={ datetimes } />
+				<TicketsList tickets={ tickets } />
+			</Fragment>;
+	}
+	return <div>{ getContent() }</div>;
+};
+
+const DatetimeAndTicketList = compose( [
+	withState( { isInitializing: true } ),
+	withSelect( ( select, ownProps ) => {
+		const { isInitializing } = ownProps;
+		const { getEntitiesForModel } = select( 'eventespresso/core' );
+		if ( isInitializing ) {
+			return ownProps;
+		}
+		return {
+			datetimes: getEntitiesForModel( 'datetime' ),
+			tickets: getEntitiesForModel( 'ticket' ),
+		};
+	} ),
+	withSelect( ( select, ownProps ) => {
+		const { eventId, isInitializing, setState } = ownProps;
+		const {
+			getEntityById,
+			getRelatedEntities,
+			getRelatedEntitiesForIds,
+		} = select( 'eventespresso/core' );
+		const { hasFinishedResolution } = select( 'core/data' );
+		if ( eventId === 0 || ! isInitializing ) {
+			return ownProps;
+		}
+		const Event = getEntityById( 'event', eventId );
+		if ( ! isModelEntityOfModel( Event, 'event' ) ) {
+			return ownProps;
+		}
+		const Datetimes = getRelatedEntities( Event, 'datetimes' );
+		const relatedIsResolved = hasFinishedResolution(
+			'eventespresso/core',
+			'getRelatedEntities',
+			[ Event, 'datetimes' ]
+		);
+		if ( ! relatedIsResolved ) {
+			return ownProps;
+		}
+		if ( ! Datetimes.length ) {
+			setState( { isInitializing: false } );
+			return ownProps;
+		}
+		const datetimeIds = Datetimes.map( ( datetime ) => datetime.id );
+		getRelatedEntitiesForIds( 'datetime', datetimeIds, 'tickets' );
+		const relatedEntitiesIsResolved = hasFinishedResolution(
+			'eventespresso/core',
+			'getRelatedEntitiesForIds',
+			[ 'datetime', datetimeIds, 'tickets' ]
+		);
+		setState( { isInitializing: false } );
+	} ),
```

Comment:
This `withSelect` HOC is responsible for populating the state with all the datetimes and tickets related to the selected EventId.  The state can be populated using the new efficient queries (getRelatedEntitiesForIds).  

The sole purpose of this HOC is to populate the state and set the `isInitializing` local state.  `isInitializing` is then used by the other hocs as a signal for the state being populated and thus other component rendering can occur from whats in the store state.

Once this HOC has populated the state, it effectively is "done", hence the early bail on line 131.  Note, this currently does not handle a change in eventId via the input (again just a raw demo).  If I was to complete this, I'd have the `onEventClick` action reset the local `isInitializing` state and clear the `eventespresso/core` state of all records from the original event.